### PR TITLE
docs: add developer guide, more concise README

### PR DESCRIPTION
**Pull request by vados-cosmonic**
# Feature or Problem
Somewhat subjective improvements to the documentation:

- add collapsible sections for install targets
- tighten copy on what wash is meant to do
- add developer guide

# Related Issues

https://github.com/wasmCloud/wash/pull/419

# Release Information

`next`

# Consumer Impact

Consumers should be able to get a feel for what `wash` is and how it fits in the ecosystem, at a glance.

# Testing

N/A

Built on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

## Unit Test(s)
N/A

## Acceptance or Integration
N/A

## Manual Verification
Yes, verified manually

### CODE REVIEW

File: docs/cli/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+## Subcommands
+
+### `wash new`
+
+Create new wasmCloud projects from predefined [templates](https://github.com/wasmCloud/project-templates). This command is a one-stop-shop for creating new actors, providers, and interfaces for all aspects of your application.
+
+### `wash up`
+
+Bootstrap a [wasmCloud][wasmcloud] environment in one easy command, supporting both launching NATS and wasmCloud in the background as well as an "interactive" mode for shorter lived hosts.
```
Comment (brooksmtownsend) [inline docs/cli/README.md]: nit: this wasmCloud link is broken, should link to the OTP repo

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
This is lovely! one nit, and conflicts that might need to be fixed

File: docs/cli/README.md (hunk 1 above)
Comment (vados-cosmonic) [reply docs/cli/README.md]: Ah this actually needs to be made consistent as well -- the link in the front README was to wasmcloud.dev -- will update them both.

**Comment by vados-cosmonic** [comment]
@brooksmtownsend thanks for the review, just patched up those issues you found! (BTW it definitely removes other people when I re-request a review)

File: README.md [hunk 2]
```diff
@@ -14,110 +14,117 @@
    \_/\_/ \__,_|___/_| |_| |_|\_____|_|\___/ \__,_|\__,_| |_____/|_| |_|\___|_|_|
 ```
 
-## Why wash
+`wash` is the comprehensive CLI to [wasmCloud][wasmcloud]. `wash` helps you:
```
Comment (brooksmtownsend) [inline README.md:17]: i'm not sure how it happens, but this link still takes me to wasmcloud.dev 🤔

File: README.md [hunk 3]
```diff
@@ -14,110 +14,117 @@
...
-## Installing wash
+`wash` is a single binary that makes developing WebAssembly with wasmCloud painless and simple.
 
-### Cargo
+## Getting started
 
-```
+<details open>
+<summary>🦀 Cargo</summary>
+The easiest way to get started with `wash` is via [`cargo`][cargo]
```
Comment (brooksmtownsend) [inline README.md:32]: This link appears to be broken, I think it can either link to Rust or just be `cargo`

File: README.md [hunk 4]
```diff
@@ -14,110 +14,117 @@
...
-Perform validation checks on .smithy models, ensuring that your interfaces are valid and usable for codegen and development.
+Feature suggestions? Find a bug? Have a question? [Submit an issue](https://github.com/wasmcloud/wash/issues/new/choose).
 
-## Contributing to wash
+Contributions of all kinds (filling issues, creating pull requests) are welcome, and the [good first issue label](https://github.com/wasmcloud/wash/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) is a great place to start.
 
-If you have any feature suggestions, find any bugs, or otherwise have a question, please submit an issue [here](https://github.com/wasmcloud/wash/issues/new/choose). Forking & submitting Pull Requests are welcome, and the [good first issue](https://github.com/wasmcloud/wash/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) label is a great way to find a place to start if you're looking to contribute.
\ No newline at end of file
+[cargo]: https://doc.rust-lang.org/cargo/
+[wasmcloud]: https://wasmcloud.dev
```
Comment (brooksmtownsend) [inline README.md]: wow, so that's how! This is a neat way to do links. But this can just change to wasmcloud.com

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
Just a request for one more broken link and .dev -> .com

File: README.md (hunk 2 above)
Comment (vados-cosmonic) [reply README.md:17]: ah I see you figured out how this works down below!

File: README.md (hunk 3 above)
Comment (vados-cosmonic) [reply README.md:32]: yeah this is my mistake here, it needs to be a HTML link since we're in `<details>`

- [x] Fix link to cargo in details

Comment (vados-cosmonic) [reply README.md:32]: Done! turns out I needed to add a newline in between to get back to regular markdown processing.

File: README.md (hunk 4 above)
Comment (vados-cosmonic) [reply README.md]: ah yeah I See you figuredout how this works, will resolve!

**Comment by vados-cosmonic** [comment]
Hey @mattwilkinsonn  any thoughts on this PR? It seems to now be conflicting with the `README.md` in `main`, but before attempting to resolve the conflicts I wanted to just get your opinion on how it should look at the end.

Here's the [README look that this PR is pushing for](https://github.com/wasmCloud/wash/blob/01acc97e9485f3a917282d665c6626ab37c54d0c/README.md).

I wanted to make the README a bit more concise around the following themes:

- A concise listing of what you can do with `wash`
- Collapsible installation instruction sections
- Only listing commands that are important to people getting started with `wash`, in the order they'd normally be useful -- `wash up`/`wash new` being first, etc.
- A separate CLI reference with all the commands and some extra direction.

Your changes added:

- A ToC at the top
- Development guide document

Which things should I keep/merge over from my README? I'm assuming the collapsible installation section stuff can stay, and *maybe* the concise listing, but are you set on having as many commands listed on the README directly? IMO commands like `par` are not something someone would normally need right away/could be tucked away.

**Pull request closed without merging**